pkg/controllers/clusterinventory/clusterprofile/controller.go

```diff
@@ -0,0 +1,265 @@
+			Type:               clusterinventory.ClusterConditionControlPlaneHealthy,
+			Status:             metav1.ConditionUnknown,
+			Reason:             clusterHealthUnknownReason,
+			ObservedGeneration: cp.Generation,
+			Message:            "The Fleet member agent has reported its status, but the health condition is missing",
+		})
+	case memberAgentLastHeartbeat == nil || time.Since(memberAgentLastHeartbeat.Time) > r.ClusterUnhealthyThreshold:
+		// The member agent has lost its heartbeat connection to the Fleet hub cluster.
+		// Set the unknown health condition in the cluster profile status.
+		meta.SetStatusCondition(&cp.Status.Conditions, metav1.Condition{
+			Type:               clusterinventory.ClusterConditionControlPlaneHealthy,
+			Status:             metav1.ConditionUnknown,
+			Reason:             clusterHeartbeatLostReason,
+			ObservedGeneration: cp.Generation,
+			Message:            "The Fleet member agent has lost its heartbeat connection to the Fleet hub cluster",
+		})
+	// TODO: Add the generation check after Fleet member agent handle the health condition appropriately.
+	case mcHealthCond.Status == metav1.ConditionUnknown:
+		// The health condition has not been updated.
+		// Set the unknown health condition in the cluster profile status.
+		meta.SetStatusCondition(&cp.Status.Conditions, metav1.Condition{
+			Type:               clusterinventory.ClusterConditionControlPlaneHealthy,
+			Status:             metav1.ConditionUnknown,
+			Reason:             clusterHealthUnknownReason,
+			ObservedGeneration: cp.Generation,
+			Message:            "The Fleet member agent health check result is out of date or unknown",
+		})
+	case mcHealthCond.Status == metav1.ConditionFalse:
+		// The member agent reports that the API server is unhealthy.
+		// Set the false health condition in the cluster profile status.
+		meta.SetStatusCondition(&cp.Status.Conditions, metav1.Condition{
+			Type:               clusterinventory.ClusterConditionControlPlaneHealthy,
+			Status:             metav1.ConditionFalse,
+			Reason:             clusterUnHealthyReason,
+			ObservedGeneration: cp.Generation,
+			Message:            "The Fleet member agent reports that the API server is unhealthy",
+		})
+	default:
+		// The member agent reports that the API server is healthy.
+		// Set the true health condition in the cluster profile status.
+		meta.SetStatusCondition(&cp.Status.Conditions, metav1.Condition{
+			Type:               clusterinventory.ClusterConditionControlPlaneHealthy,
+			Status:             metav1.ConditionTrue,
+			Reason:             clusterHealthyReason,
+			ObservedGeneration: cp.Generation,
+			Message:            "The Fleet member agent reports that the API server is healthy",
+		})
+	}
+}
+
+// cleanupClusterProfile deletes the ClusterProfile object associated with a given MemberCluster object.
+func (r *Reconciler) cleanupClusterProfile(ctx context.Context, clusterName string) error {
```

Comment:
note, here we will delete any clusterProfile no matter whether it's managed by us or not.

Reply:
I assume that we are the only manager?

Reply:
not sure if the assumption is correct or not. cause the hub is owned by customers and they could do anything.

The safe way is GET before delete.

it's kind of weird/inconsistent because in line 130 we are handling multiple managers.

Reply:
it's now an unExpected error when there are multiple managers